pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTConstructorCall.java

```diff
@@ -56,6 +58,16 @@ public boolean isQualifiedInstanceCreation() {
         return jjtGetChild(0) instanceof ASTPrimaryExpression;
     }
 
+    /**
+     * Returns the outer instance expression, if this is a {@linkplain #isQualifiedInstanceCreation() qualified}
+     * constructor call. Otherwise returns null. This can never be a
+     * {@linkplain ASTTypeExpression type expression}, and is never
+     * {@linkplain ASTAmbiguousName ambiguous}.
+     */
+    @Override
+    public @Nullable ASTExpression getQualifier() {
+        return ASTQualifiableExpression.super.getQualifier();
```

Comment:
This is just like a `super.getQualifier()` call, except unqualified `super` refers to the superclass, and the method is not inherited by the superclass, but by an interface.

Because a method that's override-equivalent to `getQualifier()` may be in the future added to any superclass, or any super interface (even one unrelated to `ASTQualifiableExpression`), the language forces us to specify from which interface exactly we want to fetch the default method. That way adding such a method to a supertype does not force us to recompile this class and change the compile-time link.

So this invokes the default method defined in ASTQualifiableExpression.

Reply:
Ok, thanks for the explanation.